File: src/libs/ThirdParty.VBSuppliers/Properties/Bonotel/IBonotelSettings.vb
Imports ThirdParty.Abstractions


Public Interface IBonotelSettings
    ReadOnly Property URL(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Username(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property Password(tpAttributeSearch As IThirdPartyAttributeSearch) As String
    ReadOnly Property AllowCancellations(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As Boolean
    ReadOnly Property OffsetCancellationDays(tpAttributeSearch As IThirdPartyAttributeSearch, IsMandatory As Boolean) As Integer
    ReadOnly Property BookTimeout(tpAttributeSearch As IThirdPartyAttributeSearch) As Integer
End Interface

File: src/libs/iVectorOne.Suppliers/Bonotel/IBonotelSettings.cs
namespace iVectorOne.CSSuppliers.Bonotel
{
    public interface IBonotelSettings
    {
        string GenericURL(IThirdPartyAttributeSearch tpAttributeSearch);
        string User(IThirdPartyAttributeSearch tpAttributeSearch);
        string Password(IThirdPartyAttributeSearch tpAttributeSearch);
        bool AllowCancellations(IThirdPartyAttributeSearch tpAttributeSearch, bool IsMandatory);
        int OffsetCancellationDays(IThirdPartyAttributeSearch tpAttributeSearch, bool IsMandatory);
        int BookTimeout(IThirdPartyAttributeSearch tpAttributeSearch);
    }
}